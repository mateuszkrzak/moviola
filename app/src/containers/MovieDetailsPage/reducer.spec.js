import {
  FETCH_MOVIE_BY_ID_REQUEST,
  FETCH_MOVIE_BY_ID_FAILURE,
  FETCH_MOVIE_BY_ID_SUCCESS,
  FETCH_MOVIES_BY_GENRES_SUCCESS,
} from './action-types';
import detailsReduer from './reducer';

describe('Details reducer', () => {
  const initialState = {
    movie: {
      title: '',
      release_date: '',
      vote_average: 0,
      poster_path: '',
      runtime: 0,
      overview: '',
      genres: [],
    },
    similarMovies: [],
    error: '',
    isLoading: false,
  };

  const mockMovie = {
    title: 'Test tile',
    release_date: '2020-10-20',
    vote_average: 9,
    poster_path: 'http://poster.path',
    runtime: 120,
    overview: 'Test overview',
    genres: ['Test genre', 'Another test genre'],
  };

  it('should return the initial state', () => {
    const action = {
      type: null,
    };

    const testedReducer = detailsReduer(undefined, action);

    expect(testedReducer).toMatchSnapshot();
  });

  it('should set isLoading after FETCH_MOVIE_BY_ID_REQUEST action', () => {
    const action = {
      type: FETCH_MOVIE_BY_ID_REQUEST,
    };

    const testedReducer = detailsReduer(initialState, action);

    expect(testedReducer).toMatchSnapshot();
  });

  it('should set movie after FETCH_MOVIE_BY_ID_SUCCESS action', () => {
    const action = {
      payload: mockMovie,
      type: FETCH_MOVIE_BY_ID_SUCCESS,
    };

    const testedReducer = detailsReduer(initialState, action);

    expect(testedReducer).toMatchSnapshot();
  });

  it('should handle FETCH_MOVIE_BY_ID_FAILURE action', () => {
    const action = {
      type: FETCH_MOVIE_BY_ID_FAILURE,
      payload: 'Test error message',
    };

    const testedReducer = detailsReduer(initialState, action);

    expect(testedReducer).toMatchSnapshot();
  });

  it('should set movie after FETCH_MOVIES_BY_GENRES_SUCCESS action', () => {
    const action = {
      payload: [mockMovie, mockMovie],
      type: FETCH_MOVIES_BY_GENRES_SUCCESS,
    };

    const testedReducer = detailsReduer(initialState, action);

    expect(testedReducer).toMatchSnapshot();
  });
});
