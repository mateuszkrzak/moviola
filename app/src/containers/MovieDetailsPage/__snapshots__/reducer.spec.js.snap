// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details reducer should handle FETCH_MOVIE_BY_ID_FAILURE action 1`] = `
Object {
  "error": "Test error message",
  "isLoading": false,
  "movie": Object {
    "genres": Array [],
    "overview": "",
    "poster_path": "",
    "release_date": "",
    "runtime": 0,
    "title": "",
    "vote_average": 0,
  },
  "similarMovies": Array [],
}
`;

exports[`Details reducer should return the initial state 1`] = `
Object {
  "error": "",
  "isLoading": false,
  "movie": Object {
    "genres": Array [],
    "overview": "",
    "poster_path": "",
    "release_date": "2018-01-30",
    "runtime": 0,
    "title": "",
    "vote_average": 0,
  },
  "similarMovies": Array [],
}
`;

exports[`Details reducer should set isLoading after FETCH_MOVIE_BY_ID_REQUEST action 1`] = `
Object {
  "error": "",
  "isLoading": true,
  "movie": Object {
    "genres": Array [],
    "overview": "",
    "poster_path": "",
    "release_date": "",
    "runtime": 0,
    "title": "",
    "vote_average": 0,
  },
  "similarMovies": Array [],
}
`;

exports[`Details reducer should set movie after FETCH_MOVIE_BY_ID_SUCCESS action 1`] = `
Object {
  "error": "",
  "isLoading": false,
  "movie": Object {
    "genres": Array [
      "Test genre",
      "Another test genre",
    ],
    "overview": "Test overview",
    "poster_path": "http://poster.path",
    "release_date": "2020-10-20",
    "runtime": 120,
    "title": "Test tile",
    "vote_average": 9,
  },
  "similarMovies": Array [],
}
`;

exports[`Details reducer should set movie after FETCH_MOVIES_BY_GENRES_SUCCESS action 1`] = `
Object {
  "error": "",
  "isLoading": false,
  "movie": Object {
    "genres": Array [],
    "overview": "",
    "poster_path": "",
    "release_date": "",
    "runtime": 0,
    "title": "",
    "vote_average": 0,
  },
  "similarMovies": Array [
    Object {
      "genres": Array [
        "Test genre",
        "Another test genre",
      ],
      "overview": "Test overview",
      "poster_path": "http://poster.path",
      "release_date": "2020-10-20",
      "runtime": 120,
      "title": "Test tile",
      "vote_average": 9,
    },
    Object {
      "genres": Array [
        "Test genre",
        "Another test genre",
      ],
      "overview": "Test overview",
      "poster_path": "http://poster.path",
      "release_date": "2020-10-20",
      "runtime": 120,
      "title": "Test tile",
      "vote_average": 9,
    },
  ],
}
`;
